proposals/lip-0042.md

```diff
@@ -134,11 +173,19 @@ The default reward of the block as a 64-bit unsigned integer.
 
 #### After Block Execution
 
-After a block `b` is executed, the `mint` function exposed by the [token module][research:token-module] to assign the reward to the block generator as:
+After a block `b` is executed, the `mint` function exposed by the [Token module][token-module] to assign the reward to the block generator as:
 
 ```python
-blockReward = getBlockReward(b.header)
+(blockReward, reduction) = getBlockReward(b.header)
 mint(b.header.generatorAddress, TOKEN_ID_REWARD, blockReward)
+
+eventDataBytes = encode(
+schema = rewardMintedDataSchema,
+object = {
+    "amount": blockReward,
+    "reduction": reduction
+})
+emitEvent(MODULE_ID_REWARD, TYPE_ID_REWARD_MINTED, eventDataBytes, [b.header.generatorAddress])
```

Comment:
check the new API, we don't need to encode before anymore (see PR of 44 as reference)

Reply:
Super cool API!